ocaml/xapi/xapi_pvs_proxy.ml

```diff
@@ -33,7 +33,13 @@ let create ~__context ~site ~vIF ~prepopulate =
 	if Db.VIF.get_currently_attached ~__context ~self:vIF then begin
 		Pvs_proxy_control.start_proxy ~__context vIF pvs_proxy;
 		if Db.PVS_proxy.get_currently_attached ~__context ~self:pvs_proxy then
-			Xapi_xenops.vif_set_pvs_proxy ~__context ~self:vIF true
+			try Xapi_xenops.vif_set_pvs_proxy ~__context ~self:vIF true
+			with e ->
+				let body = Printf.sprintf "Failed to setup pvs proxy for vif:%s"
+					(Db.VIF.get_uuid ~__context ~self:vIF) in
+				let (name, priority) = Api_messages.pvs_proxy_setup_failed in
+				ignore(Xapi_message.create ~__context ~name ~priority ~cls:`PVS_proxy ~obj_uuid:uuid ~body);
```

Comment:
This won't work well on a pool slave: use the API through the Client module (`Client.Message.create` etc).